### PR TITLE
V6/swap mewwallet

### PR DESCRIPTION
**Pull request by gamalielhere**
### Devop

* \[x] Updated CHANGELOG.md
* \[x] Add PR label
* \[x] No hard-coded strings

### CODE REVIEW

**Comment by olgakup** [comment]
It is still not working correctly, once the user has aproved transaction on the mew wallet( there is no action required from the user on the web) it should be sent automatically and they should see swap initiated message

**Comment by olgakup** [comment]
Also if the user cancels the transaction on mew wallet, the ui on V6 still shows 'Approve 1 transaction on MEW Wallet.', while it should show error message and the button confirm and send shoudl be enabled

File: src/modules/confirmation/ModuleConfirmation.vue [hunk 1]
```diff
@@ -334,9 +334,12 @@ export default {
     isWeb3Wallet() {
       return this.identifier === WALLET_TYPES.WEB3_WALLET;
     },
+    isMewConnect() {
+      return this.identifier === WALLET_TYPES.MEW_CONNECT;
+    },
     showConfirmWithWallet() {
       return (
-        (this.isHardware || this.isWeb3Wallet) &&
+        (this.isHardware || this.isWeb3Wallet || this.isMewConnect) &&
```
Comment (olgakup) [inline src/modules/confirmation/ModuleConfirmation.vue:342]: does isHardware covers other mobile protocols?

**Review by olgakup — DISMISSED** [review]
Please see comments on the pr

File: src/modules/confirmation/ModuleConfirmation.vue [hunk 2]
```diff
@@ -528,6 +556,7 @@ export default {
     EventBus.$on(
       EventNames.SHOW_BATCH_TX_MODAL,
       async (arr, resolver, isHardware) => {
+        console.log('supposed to be here', arr);
```
Comment (gesseekur) [inline src/modules/confirmation/ModuleConfirmation.vue:559]: console! unless its supposed to be here lolol

File: src/modules/confirmation/components/ConfirmationSwapTransactionDetails.vue [hunk 3]
```diff
@@ -139,21 +147,24 @@ export default {
...
         },
         {
           title: 'You will get',
           icon: this.toImg,
           value: this.formattedToVal,
-          type: this.toType
+          type: this.toType,
+          address: this.to,
+          amount: formatFloatingPointValue(this.toVal).value,
+          usd: formatFloatingPointValue(this.toUsd).value
```
Comment (gesseekur) [inline src/modules/confirmation/components/ConfirmationSwapTransactionDetails.vue:162]: should usd use formatFiatValue instead ?

File: src/modules/swap/ModuleSwap.vue [hunk 4]
```diff
@@ -1080,6 +1084,7 @@ export default {
     },
 
     executeTrade() {
+      console.log(this.confirmInfo);
```
Comment (gesseekur) [inline src/modules/swap/ModuleSwap.vue:1087]: consoleeee

File: src/modules/confirmation/ModuleConfirmation.vue (hunk 2 above)
Comment (kvhnuke) [reply src/modules/confirmation/ModuleConfirmation.vue:559]: consooooolleeeeee

File: src/modules/confirmation/ModuleConfirmation.vue [hunk 5]
```diff
@@ -728,17 +757,26 @@ export default {
...
             this.signedTxObject = res;
+            if (this.isHardware && this.txSigned) {
+              this.btnAction();
+            }
           })
           .catch(e => {
             this.signedTxObject = {};
-            this.error = e.message;
+            if (this.isMewConnect) {
+              this.error = 'User declined action';
```
Comment (kvhnuke) [inline src/modules/confirmation/ModuleConfirmation.vue:775]: cant we get the error from MEWconnect?

**Review by kvhnuke — CHANGES_REQUESTED** [review]

**Review by kvhnuke — APPROVED** [review]

**Comment by kvhnuke** [comment]
> It is still not working correctly, once the user has aproved transaction on the mew wallet( there is no action required from the user on the web) it should be sent automatically and they should see swap initiated message

this actually the case for all harware wallets, Ill fix it tomorrow

**Review by kvhnuke — APPROVED** [review]

**Review by gesseekur — APPROVED** [review]

**Pull request merged**